main.go

```diff
@@ -251,34 +102,42 @@ func main() {
 
 	cacheRecorderReader.Restore()
 
-	currentStackID := strings.TrimSpace(conf.StackID)
-	if len(currentStackID) > 0 {
+	currentStackInfo := model.ArchiveInfo{
+		StackID:      strings.TrimSpace(conf.StackID),
+		Architecture: currentArchitecture,
+	}
+	if len(currentStackInfo.StackID) > 0 {
 		fmt.Println()
 		log.Infof("Checking archive and current stacks")
-		log.Printf("current stack id: %s", currentStackID)
+		log.Printf("current stack: %s", currentStackInfo)
 
 		if filepath.Base(hdr.Name) == "archive_info.json" {
 			b, err := ioutil.ReadAll(r)
 			if err != nil {
 				failf("Failed to read first archive entry: %s", err)
 			}
 
-			archiveStackID, err := parseStackID(b)
+			archiveStackInfo, err := parseArchiveInfo(b)
 			if err != nil {
 				failf("Failed to parse first archive entry: %s", err)
 			}
-			log.Printf("archive stack id: %s", archiveStackID)
+			log.Printf("archive stack: %s", archiveStackInfo)
 
-			if !isSameStack(archiveStackID, currentStackID) {
-				log.Warnf("Cache was created on stack: %s, current stack: %s", archiveStackID, currentStackID)
-				log.Warnf("Skipping cache pull, because of the stack has changed")
+			if !isSameStack(archiveStackInfo, currentStackInfo) {
+				log.Warnf("Cache was created on stack: %s, current stack: %s", archiveStackInfo, currentStackInfo)
```

Comment:
`isSameStack()` checks both the stack AND architecture, while this error message gives the stack ID mismatch as the reason for skipping the cache pull.

Reply:
It prints architecture too.

Reply:
I checked the `String()` implementation of `ArchiveInfo`, you are right, gotcha!